pkg/kubelet/stats/cri_stats_provider.go

```diff
@@ -826,7 +826,29 @@ func getCRICadvisorStats(infos map[string]cadvisorapiv2.ContainerInfo) map[strin
 		if !isPodManagedContainer(&info) {
 			continue
 		}
-		stats[path.Base(key)] = info
+		stats[extractIDFromCgroupPath(key)] = info
 	}
 	return stats
 }
+
+func extractIDFromCgroupPath(cgroupPath string) string {
+	//case0: /kubepods/burstable/pod2fc932ce-fdcc-454b-97bd-aadfdeb4c340/9be25294016e2dc0340dd605ce1f57b492039b267a6a618a7ad2a7a58a740f32
+	id := path.Base(cgroupPath)
+
+	//case1: "/kubepods.slice/kubepods-burstable.slice/kubepods-burstable-pod2fc932ce_fdcc_454b_97bd_aadfdeb4c340.slice/cri-containerd-aaefb9d8feed2d453b543f6d928cede7a4dbefa6a0ae7c9b990dd234c56e93b9.scope"
+	const scopeSuffix string = ".scope"
+	if strings.HasSuffix(id, scopeSuffix) {
+		id = strings.TrimSuffix(id, scopeSuffix)
+	}
+	components := strings.Split(id, "-")
+	if len(components) > 1 {
+		id = components[len(components)-1]
+	}
+
+	//case2: "/kubepods-burstable-pod2fc932ce_fdcc_454b_97bd_aadfdeb4c340.slice:cri-containerd:db622e6f7b16079db56fd7814031f7e8fc78b3050c24fe2fc431beeebbeb1c8e"
```

Comment:
I'm a bit surprised by the format here. I was under previous assumption that the cgroup name should end in ".slice" as this is what  the `IsSystemdStyleName` logic has: https://github.com/kubernetes/kubernetes/blob/5be21c50c269fc1d28e0bd31ab9dcb572ae7fac5/pkg/kubelet/cm/cgroup_manager_linux.go#L132:6

/cc @haircommander @mrunalp who have more familiarity with systemd cgroup driver

Reply:
@bobbypage  For the `.slice` suffix,  I think `IsSystemdStyleName` might only apply to PodID instead of ContainerID.
Here's an example of piece of `allInfos`
```
        "/kubepods-besteffort-pod172f50f4_10ce_4f06_af33_3e1f0716b2a0.slice:cri-containerd:fa98e6c3491302833bd4b68c9ca866569cc40e7858d5881501e84b4dad29e3bb": *{
                Spec: (*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerSpec")(0xc0023c97a0),
                Stats: []*k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats len: 2, cap: 2, [
                        *(*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats")(0xc0024070e0),
                        *(*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats")(0xc0024071d0),
                ],}, 
        "/kubepods.slice/kubepods-burstable.slice": *{
                Spec: (*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerSpec")(0xc0023c99e0),
                Stats: []*k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats len: 2, cap: 2, [
                        *(*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats")(0xc0024072c0),
                        *(*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats")(0xc0024073b0),
                ],}, 
        "/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod172f50f4_10ce_4f06_af33_3e1f0716b2a0.slice": *{
                Spec: (*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerSpec")(0xc0023c9320),
                Stats: []*k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats len: 2, cap: 2, [
                        *(*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats")(0xc002406d20),
                        *(*"k8s.io/kubernetes/vendor/github.com/google/cadvisor/info/v2.ContainerStats")(0xc002406e10),
                ],}, 
```

Reply:
yeah I believe IsSystemdStyleName applies to the pod scope. Underneath the pod scope are the container slices. Interestingly, case 2 is the way the kubelet specifies to the CRI for a systemd scope. I believe the path always ends up being `.slice`. I would not expect the colon separated systemd slice description to leak to the cgroup heierarchy that cadvisor reads. @YanzhaoLi did you notice this happening?

Reply:
`I believe the path always ends up being .slice (as in case 1)`: the pod path always ends up being `.slice` while the containerd path not. (In case1, it ends up being `.scope`)

` I would not expect the colon separated systemd slice description to leak to the cgroup heierarchy that cadvisor reads`: cadvisor can read all info correctly, but when storing the infos to map, it cannot extract the real containerID from systemd-style path https://github.com/kubernetes/kubernetes/blob/d4179be6117c8a17e4c22f3bd7f8aede407df440/pkg/kubelet/stats/cri_stats_provider.go#L814